code/midtown/mmai/aiVehicleOpponent.cpp

```diff
@@ -20,5 +20,63 @@ define_dummy_symbol(mmai_aiVehicleOpponent);
 
 #include "aiVehicleOpponent.h"
 
+#include "agi/dlptmpl.h"
+#include "agi/getdlp.h"
+#include "agiworld/quality.h"
+
+#include "aiData.h"
+#include "aiGoalBackup.h"
+#include "aiGoalFollowWayPts.h"
+#include "aiGoalStop.h"
+
 void aiVehicleOpponent::DrawDamage()
 {}
+
+void aiVehicleOpponent::Init(int opp_id, aiRaceData* race_data, char* arg3)
+{
+    int TextureQuality = agiRQ.TextureQuality;
+    if (agiRQ.TextureQuality)
+        --agiRQ.TextureQuality;
+
+    int paint_job = opp_id & 3;
+    int index = opp_id + 1;
+
+    List* p_opponents = &race_data->Opponents;
+    OpponentRaceData* oppRaceData = static_cast<OpponentRaceData*>(p_opponents->Access(index));
+
+    Car.Init(oppRaceData->Model, CAR_TYPE_OPPONENT, paint_job);
+
+    agiRQ.TextureQuality = TextureQuality;
+
+    aiVehicle::Init(opp_id);
+    Car.Reset();
+
+    std::unique_ptr<aiGoalFollowWayPts> followWayPts = std::make_unique<aiGoalFollowWayPts>(
+        oppRaceData->PathFile, &RailSet, this, &IsBackup, &IsFinished, &IsStopped, arg3, oppRaceData->MaxThrottle);
+
+    WayPts = followWayPts.release();
```

Comment:
Instead of converting back into raw pointers, it should be safe to set the members to `Ptr<T>` instead